src/storage/lock_manager.rs

```diff
@@ -1,13 +1,51 @@
 // Copyright 2019 TiKV Project Authors. Licensed under Apache-2.0.
 
-use crate::storage::{types::ProcessResult, StorageCallback, TimeStamp};
+use crate::storage::{txn::ProcessResult, types::StorageCallback};
+use keys::TimeStamp;
 
 #[derive(Clone, Copy, PartialEq, Debug, Default)]
 pub struct Lock {
     pub ts: TimeStamp,
     pub hash: u64,
 }
 
+/// Time to wait for lock released when encountering locks.
+#[derive(Clone, Copy, PartialEq, Debug)]
+pub enum WaitTimeout {
+    None,
+    Default,
```

Comment:
Could we remove the `Default` option and implement default for WaitTimeout which means WaitTimeout(0) ?

Reply:
The idea of using an enum here is to make it explicit when we are using a default value and when we are using a specified value, making `0` mean default is making the difference implicit and therefore harder to understand.